Core/FileLoader.swift

```diff
@@ -27,15 +27,22 @@ enum FileError: Error {
 
 class FileLoader {
 
-    func load(name: String, ext: String) throws -> Data {
+    func load(fileName: String) throws -> Data {
         let bundle = Bundle.init(for: FileLoader.self)
-        return try load(bundle: bundle, name: name, ext: ext)
+        return try load(fileName: fileName, fromBundle: bundle)
     }
     
-    func load(bundle: Bundle, name: String, ext: String) throws -> Data {
-        guard let path = bundle.path(forResource: name, ofType: ext) else { throw  FileError.unknownFile }
-        guard let string = try? String(contentsOfFile: path) else { throw FileError.invalidFileContents }
-        guard let data = string.data(using: .utf16) else { throw FileError.invalidFileContents }
+    func load(fileName: String, fromBundle bundle: Bundle) throws -> Data {
+        
+        let fileUrl = URL(fileURLWithPath: fileName)
+        let baseName = fileUrl.deletingPathExtension().path
+        let ext = fileUrl.pathExtension
```

Comment:
Seems excessive when you could just pass the extension in, but I understand the motivation.

Reply:
Caine gave me the opposite feedback - I sat on it for ages and finally put it in :-)

Reply:
I'm not that fussed, but passing the extension is no surprise to an iOS dev and having to construct a URL just to get the extension (which is probably slightly less offensive than using a regex ;) ) still fees a bit OTT to me :)

Happy for it to stay.